File: src/main/java/com/life/premium/domain/PremiumCalculationResult.java
package com.life.premium.domain;

/**
 * 保費試算結果值物件（persistence-free）
 */
public class PremiumCalculationResult {
    private final long annualPremium;
    private final long monthlyPremium;
    private final String currency;

    public PremiumCalculationResult(long annualPremium, long monthlyPremium, String currency) {
        this.annualPremium = annualPremium;
        this.monthlyPremium = monthlyPremium;
        this.currency = currency;
    }

    public long getAnnualPremium() { return annualPremium; }
    public long getMonthlyPremium() { return monthlyPremium; }
    public String getCurrency() { return currency; }
}

File: src/LifePremium.Domain/Entities/PremiumCalculationResult.cs
namespace LifePremium.Domain.Entities;

/// <summary>
/// 保費試算結果 (BR-004: 年繳保費, BR-005: 月繳保費)
/// </summary>
public sealed record PremiumCalculationResult
{
    public decimal AnnualPremium { get; }
    public decimal MonthlyPremium { get; }

    public PremiumCalculationResult(decimal annualPremium, decimal monthlyPremium)
    {
        AnnualPremium = annualPremium;
        MonthlyPremium = monthlyPremium;
    }
}
